src/bin/smoosh_tools.rs

```diff
@@ -0,0 +1,787 @@
+
+    update_cargo(
+        &mc.smoosh_cargo,
+        LocalDependencyLineReplacer {
+            jsparagus: jsparagus.topsrcdir,
+        },
+    )?;
+
+    run_mach("build", args)
+}
+
+fn shell(args: &SimpleArgs) -> Result<(), Error> {
+    run_mach("run", args)
+}
+
+fn test(args: &SimpleArgs) -> Result<(), Error> {
+    run_mach("jstests", args)?;
+    run_mach("jit-test", args)
+}
+
+fn vendor(mc: &MozillaCentralTree) -> Result<(), Error> {
+    let args = &["vendor", "rust"];
+    let cmdline = format_cmdline("mach", args);
+    log_info_cmd!(cmdline);
+    check_command(
+        Command::new(mc.topsrcdir.join("mach").to_str().unwrap())
+            .args(args)
+            .current_dir(mc.topsrcdir.clone()),
+        cmdline,
+    )
+}
+
+fn bump(args: &SimpleArgs) -> Result<(), Error> {
+    let mc = MozillaCentralTree::try_new(&args.mc_path)?;
+    let jsparagus = JsparagusTree::try_new()?;
+
+    let jsparagus_repo = GitRepository::try_new(jsparagus.topsrcdir.clone())?;
+
+    log_info!("Checking ci_generated branch HEAD");
+
+    jsparagus_repo.ensure_remote(
+        "upstream",
+        "https://github.com/mozilla-spidermonkey/jsparagus.git",
+    )?;
+    let remotes = jsparagus_repo.ls_remote("upstream")?;
+
+    let branch = "refs/heads/ci_generated";
+
+    let ci_generated_sha = match remotes.get(branch) {
+        Some(sha) => sha,
+        None => {
+            return Err(Error::Generic(format!("{} not found in upstream", branch)));
+        }
+    };
+
+    log_info!("ci_generated branch HEAD = {}", ci_generated_sha);
+
+    update_cargo(
+        &mc.smoosh_cargo,
+        OfficialDependencyLineReplacer {
+            sha: ci_generated_sha.clone(),
+        },
+    )?;
+
+    vendor(&mc)?;
+
+    log_info!("Please add updated files and commit them.");
+
+    Ok(())
+}
+
+/// Parse remote string and get GitHub username.
+/// Currently this supports only SSH format.
+fn parse_github_username(remote: String) -> Result<String, Error> {
+    let git_prefix = "git@github.com:";
+    let git_suffix = "/jsparagus.git";
+
+    if remote.starts_with(git_prefix) && remote.ends_with(git_suffix) {
+        return Ok(remote.replace(git_prefix, "").replace(git_suffix, ""));
+    }
+
+    Err(Error::Generic(format!(
+        "Failed to get GitHub username: {}",
+        remote
+    )))
+}
+
+struct BranchInfo {
+    github_user: String,
+    branch: String,
+}
+
+/// Create "generated" branch and push to remote, and returns
+/// GitHub username and branch name.
+fn push_to_gen_branch() -> Result<BranchInfo, Error> {
+    let jsparagus = JsparagusTree::try_new()?;
+
+    let jsparagus_repo = GitRepository::try_new(jsparagus.topsrcdir.clone())?;
+    jsparagus_repo.assert_clean()?;
+
+    log_info!("Getting GitHub username and current branch");
+
+    let origin = jsparagus_repo
+        .get_output(&["remote", "get-url", "origin"])?
+        .trim()
+        .to_string();
+    let github_user = parse_github_username(origin)?;
```

Comment:
This code assumes that `origin` is the user repository, However, this is not garanteed. Running `git remote -v` locally results in:
```
github	git@github.com:nbp/jsparagus.git (fetch)
github	git@github.com:nbp/jsparagus.git (push)
jorendorff	https://github.com/jorendorff/jsparagus.git (fetch)
jorendorff	https://github.com/jorendorff/jsparagus.git (push)
origin	https://github.com/mozilla-spidermonkey/jsparagus.git (fetch)
origin	git@github.com:mozilla-spidermonkey/jsparagus.git (push)
```

If I were to use this tool, it would publish my changes to `mozilla-spidermonkey/jsparagus.git` instead of `nbp/jsparagus.git`.

I do not think there is any solution capable of guessing which remote to use, knowing that one can push using https as well. You might consider taking the remote name as argument.

Reply:
added `--remote=REMOTE` argument